File: Backend/SoftAseg/SoftAsegBusiness/src/main/java/pe/edu/pucp/softaseg/bo/VehiculoBO.java

package pe.edu.pucp.softaseg.bo;

import java.util.ArrayList;
import pe.edu.pucp.softaseg.dao.VehiculoDAO;
import pe.edu.pucp.softaseg.daoImp.VehiculoDAOImpl;
import pe.edu.pucp.softaseg.model.gestionSiniestros.VehiculosDTO;

public class VehiculoBO {
    private VehiculoDAO vehiculoDAO;
    
    public VehiculoBO(){
        this.vehiculoDAO = new VehiculoDAOImpl();
    }
    
    public Integer insertar(VehiculosDTO vehiculosDTO){
        Integer vehiculoId = this.vehiculoDAO.insertar(vehiculosDTO);
        vehiculosDTO.setVehiculoId(vehiculoId);
        return vehiculoId;
    }
    
    public VehiculosDTO obtenerPorId(Integer vehiculoId){
        return this.vehiculoDAO.obtenerPorId(vehiculoId);
    }
    
    public ArrayList<VehiculosDTO> listarTodos(){
        return this.vehiculoDAO.listarTodos();
    }
    
    public ArrayList<VehiculosDTO> listarPorPlacaMarcaModeloTipoVehiculo(String filtro) {
        return this.vehiculoDAO.listarPorPlacaMarcaModeloTipoVehiculo(filtro);
    }
    
    public Integer modificar(VehiculosDTO vehiculosDTO){
        return this.vehiculoDAO.modificar(vehiculosDTO);
    }
    
    public Integer eliminar(Integer vehiculoId){
        VehiculosDTO vehiculosDTO = new VehiculosDTO();
        vehiculosDTO.setVehiculoId(vehiculoId);
        return this.vehiculoDAO.eliminar(vehiculosDTO);
    }
}

File: Frontend/SoftSiniNet/SiniNetBusiness/VehiculoBO.cs
using SiniNetBusiness.SiniNetWSVehiculos;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SiniNetBusiness
{
    public class VehiculoBO
    {
        private VehiculosClient clienteSOAP;

        public VehiculoBO()
        {
            this.clienteSOAP = new VehiculosClient();
        }

        public int insertar(string placa, marcaVehiculo marca, string modelo, string color, tipoVehiculo tipo)
        {
            return this.clienteSOAP.insertarVehiculos(placa, marca, modelo, color, tipo);
        }

        public vehiculosDTO obtenerPorId(int vehiculoId)
        {
            return this.clienteSOAP.obtenerPorIdVehiculos(vehiculoId);
        }

        public IList<vehiculosDTO> listarTodos()
        {
            return this.clienteSOAP.listarTodosVehiculos();
        }

        public IList<vehiculosDTO> listarPorPlacaMarcaModeloTipoVehiculo(string filtro)
        {
            return this.clienteSOAP.listarPorPlacaMarcaModeloTipoVehiculo(filtro);
        }

        public int modificar(int vehiculoId, string placa, marcaVehiculo marca, string modelo, string color, tipoVehiculo tipo)
        {
            return this.clienteSOAP.modificarVehiculos(vehiculoId, placa, marca, modelo, color, tipo);
        }

        public int eliminar(int vehiculoId)
        {
            return this.clienteSOAP.eliminarVehiculos(vehiculoId);
        }
    }

}
